howto/diagnose-with-tracing/open-telemetry-collector.md

```diff
@@ -0,0 +1,254 @@
+    extensions:
+      health_check:
+      pprof:
+        endpoint: :1888
+      zpages:
+        endpoint: :55679
+    exporters:
+      azuremonitor:
+      azuremonitor/2:
+        # Configuring OpenTelemetry Collector to send data to your
+        # Application Insights instance
+        endpoint: "https://dc.services.visualstudio.com/v2/track"
+        instrumentation_key: "$APP_INSIGHT_KEY"
+        # maxbatchsize is the maximum number of items that can be queued
+        # before calling to the configured endpoint
+        maxbatchsize: 100
+        # maxbatchinterval is the maximum time to wait before calling
+        # the configured endpoint.
+        maxbatchinterval: 10s
+    service:
+      extensions: [pprof, zpages, health_check]
+      pipelines:
+        traces:
+          receivers: [opencensus]
+          exporters: [azuremonitor/2]
+          processors: [batch, queued_retry]
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: otel-collector
+  labels:
+    app: opencesus
+    component: otel-collector
+spec:
+  ports:
+  - name: opencensus # Default endpoint for Opencensus receiver.
+    port: 55678
+    protocol: TCP
+    targetPort: 55678
+  selector:
+    component: otel-collector
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: otel-collector
+  labels:
+    app: opentelemetry
+    component: otel-collector
+spec:
+  replicas: 1  # scale out based on your usage
+  selector:
+    matchLabels:
+      app: opentelemetry
+  template:
+    metadata:
+      labels:
+        app: opentelemetry
+        component: otel-collector
+    spec:
+      containers:
+      - name: otel-collector
+        image: otel/opentelemetry-collector-contrib-dev:latest
+        command:
+          - "/otelcontribcol"
+          - "--config=/conf/otel-collector-config.yaml"
+        resources:
+          limits:
+            cpu: 1
+            memory: 2Gi
+          requests:
+            cpu: 200m
+            memory: 400Mi
+        ports:
+          - containerPort: 55678 # Default endpoint for Opencensus receiver.
+        volumeMounts:
+          - name: otel-collector-config-vol
+            mountPath: /conf
+          #- name: otel-collector-secrets
+          #  mountPath: /secrets
+        livenessProbe:
+          httpGet:
+            path: /
+            port: 13133
+        readinessProbe:
+          httpGet:
+            path: /
+            port: 13133
+      volumes:
+        - configMap:
+            name: otel-collector-conf
+            items:
+              - key: otel-collector-config
+                path: otel-collector-config.yaml
+          name: otel-collector-config-vol
+EOF
+
+```
+
+
+Set up a Dapr exporter component
+```
+kubectl apply -f- <<EOF
+apiVersion: dapr.io/v1alpha1
+kind: Configuration
+metadata:
+  name: appconfig
+  namespace: default
+spec:
+  # Configure Dapr tracing.
+  tracing:
+    samplingRate: "1"
+---
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: native
```

Comment:
Is the component type really "native". This seems and odd name.
@youngbupark - Where did this name come from and should there be a better name?

Reply:
Yes, the component is really named `native`... it is a misnomer I believe.